### PR TITLE
MAILET-166 AutomaticSentMailDetectorImpl should not fail above 1MB

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailet/base/src/test/java/org/apache/mailet/base/AutomaticallySentMailDetectorImplTest.java [hunk 1]
```diff
@@ -332,4 +333,23 @@ public void isMdnSentAutomaticallyShouldNotThrowOnBodyPartsWithManyLines() throw
         assertThat(new AutomaticallySentMailDetectorImpl().isMdnSentAutomatically(fakeMail)).isFalse();
     }
 
+    @Test
+    public void isMdnSentAutomaticallyShouldNotThrowWhenBiggerThan1MB() throws Exception {
+        MimeMessageBuilder message = MimeMessageBuilder.mimeMessageBuilder()
+            .addHeaders()
+            .setMultipartWithBodyParts(
+                MimeMessageBuilder.bodyPartBuilder()
+                    .data(Strings.repeat("12345678\r\n", 150 * 1024)),
```
Comment (mbaechler) [inline mailet/base/src/test/java/org/apache/mailet/base/AutomaticallySentMailDetectorImplTest.java:342]: with jdk 11, you have 
```
"12345678\r\n".repeat(150 * 1024)
```

**Comment by blackheaven** [comment]
Should not we increase the size? or is it in the spec?

**Comment by rouazana** [comment]
> Should not we increase the size? or is it in the spec?

That's a good question: as the MDN can optionally include the original message, there is no reason to expect the size to be limited.

**Comment by chibenwa** [comment]
```
[c7d6b1deb3946e0a1d267140cebd6608616eda02] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project memory-webadmin-integration-test: There was a timeout or other error in the fork -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
> That's a good question: as the MDN can optionally include the original message, there is no reason to expect the size to be limited.

I'm OK for increasing the stream limit to let's say 100 MB

**Comment by chibenwa** [comment]
> It lacks of a proper test

I wrote a proper test.

Now please be helpful, tell me which kind of test you want.

If it is too complicated, I turn this into a backlog ticket and we handle it next sprint.

**Comment by blackheaven** [comment]
A test with a valid MDN over 100 MiB

**Comment by chibenwa** [comment]
> A test with a valid MDN over 100 MiB

I'm not writing such a test sorry. I don't see it's value.

**Comment by mbaechler** [comment]
> > Should not we increase the size? or is it in the spec?
> 
> That's a good question: as the MDN can optionally include the original message, there is no reason to expect the size to be limited.

How is it included? Does a parsing limited to first mime level could identify if it's an MDN without parsing 100MiB?

**Comment by blackheaven** [comment]
> I'm not writing such a test sorry. I don't see its value.

Sure, what about a 2 MiB MDN then?

**Comment by chibenwa** [comment]
> How is it included? Does a parsing limited to first mime level could identify if it's an MDN without parsing 100MiB?

The algorithm is currently "" does the mail have any mime part with message/disposition-notification "".

Given this algorithm you need to consume the stream until the end of the mime content.

> Sure, what about a 2 MiB MDN then?

Sounds doable then.

**Comment by mbaechler** [comment]
Merged

**Pull request closed without merging**